### DISCUSSION TITLE
useContractWrite() write is undefined but `isError==false`

### DISCUSSION THREAD

**Discussion by gruvin** [wevm/wagmi#2576] [Q&A] [answered]
Wagmi first-timer. `v1.1.1` according to `package.json` using `React`.

I've encountered an instance where `useContractWrite(config)` is returning `write=undefined` while also `isError=false` and `data=undefined`.

I _think_ the issue _might_ be that the wallet (Metamask) is returning some error thus preventing Viem's `simulateTransaction()` from seeing an issue? Perhaps something like, "insufficient funds for gas" -- though there appears to be plenty and I see no other reasons the transaction request should fail. Either way, I cannot find a way to see what the issue actually is.

I'm using `const { config } = userPrepareContractWrite()` prior to this `useContractWrite()` and that also returns no error. I wonder if Wagmi doesn't expose all of Viem's `simulateContract()` return values and perhaps maybe that's why I cannot see what is happening behind the scenes?

```javascript
    const { data: pdata, config } = usePrepareContractWrite({
        address: Contract.Address,
        abi: Contract.ABI,
        functionName: "stakeEnd",
        args: [props.index, Number(props.id)],
        onError: (e) => debug("ERROR PrepareContractWrite: ", e)
    })
    debug("PrepareContractWrite: ", pdata, config)

    const { data, isLoading, isSuccess, isError, reset, write } = useContractWrite(config)
    debug("ContractWrite: ", { data, isLoading, isSuccess, isError, reset, write })
```

With no `write()` to call, I haven't the opportunity to involve Metamask's UI for further hints.

The contract itself is not mine and has been around for a few years with no reports of trouble.

How might I debug this to see what is going on in the background? Or is the problem more obvious to you, perhaps? ;-)

Thanks!

**Comment by jxom** [comment]
Is there an error on the `usePrepareContractWrite` Hook? You can check by logging `error` on `usePrepareContractWrite` return value.

**Comment by gruvin** [comment]
Thanks for the reply @jxom. Nope. No errors anywhere.

```javascript
const { data: pdata, error: perror, config } = usePrepareContractWrite({
...
debug("PDATA: ", pdata, perror) // => PDATA:  undefined null
```

**Reply by jxom** [reply, marked as answer]
Hmm, looks like your Prepare Hook isn't running (ie. you're not getting anything back). The Prepare Hook will only run when:

- You have a defined `abi` ✅
- You have a defined `address` ✅
- You have a defined `functionName` ✅
- Your wallet is connected ❓

Is your wallet connected?

**Reply by gruvin** [reply]
Oh. Geeze. I believe you are right. I have an `enabled: ...` clause in the actual code with bad logic.
